src/dstore/dstore_base.c

```diff
@@ -799,3 +757,125 @@ int dstore_pread(struct dstore_obj *obj, off_t offset, size_t count,
 
 	return rc;
 }
+
+int dstore_dealloc_op(struct dstore_obj *obj, struct dstore_io_vec *vec,
+		      struct dstore_io_op **out)
+{
+	int rc;
+
+	rc = dstore_io_op_init_and_submit(obj, vec, out,
+					  DSTORE_IO_OP_FREE);
+
+	log_debug("dstore_dealloc_op (" OBJ_ID_F " <=> %p, "
+		  "vec=%p *out=%p) rc=%d",
+		  OBJ_ID_P(dstore_obj_id(obj)), obj, vec,
+		  rc == 0 ? *out : NULL, rc);
+
+	return rc;
+}
+
+int dstore_deallocate(struct dstore_obj *obj, off_t offset, size_t count)
+{
+	int rc = 0;
+
+	dassert(obj);
+
+	char *tmp_buf = NULL;
+	struct dstore_io_op *wop = NULL;
+	struct dstore_io_vec *vec = NULL;
+	size_t nr_request;
+	size_t tail_size;
+	size_t nblk_per_req;
+	size_t ndata_per_req;
+	int i;
+
+	size_t bsize = dstore_get_bsize(obj->ds,
+					(dstore_oid_t *)dstore_obj_id(obj));
+
+	if (offset % bsize != 0) {
+		/*we need to make deallocate operation left aligned */
+		tmp_buf = calloc(bsize, sizeof(char));
+		uint32_t left_blk_num = offset/bsize;
+		uint32_t write_count = offset - (left_blk_num * bsize);
+		write_count = bsize - write_count;
+
+		if (count < write_count) {
+
+			write_count = count;
+		}
+
+		RC_WRAP_LABEL(rc, out, dstore_pwrite, obj,
+			      offset, write_count,
+			      bsize, tmp_buf);
+
+		if (write_count == count) {
+			goto out;
+		}
+
+		count = count - write_count;
+		offset = offset + write_count;
+	}
+
+	ndata_per_req = DSAL_MAX_IO_SIZE;	
```

Comment:
As mentioned in earlier comment division should happen with 20MB so less number of truncate request to Motr.

Reply:
Done.